Use double-backticks in Web Extension Headers docs for symbols and methods.

xeenon opened the pull request:
#### 64dcfec215ad425f82292aa591c6c94554641cba
<pre>
Use double-backticks in Web Extension Headers docs for symbols and methods.
<a href="https://webkit.org/b/278174">https://webkit.org/b/278174</a>
<a href="https://rdar.apple.com/problem/133961517">rdar://problem/133961517</a>

Reviewed by Jeff Miller.

Using double backticks is recommended for symbols and methods, and are highlighted
differently in Xcode, and developers can jump to the definition with command-click.

* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataRecord.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataType.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPermission.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.h:

Canonical link: <a href="https://commits.webkit.org/282305@main">https://commits.webkit.org/282305@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb7f61b2ce47558de3f0dac722930a7ce395f5ff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62775 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42131 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15371 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/66795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/13379 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/49818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13663 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/66795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/13379 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65844 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/49818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/54381 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/66795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/49818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/12255 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/49818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/12042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/68490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/87/builds/6721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/11695 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/68490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/86/builds/6753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/54434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/68490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9453 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/37951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/39031 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/40142 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/38773 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

jeffmapple commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
```diff
@@ -62,12 +62,12 @@ typedef NS_ERROR_ENUM(WKWebExtensionErrorDomain, WKWebExtensionError) {
 
 /*!
- @abstract A `WKWebExtension` object encapsulates a web extension’s resources that are defined by a `manifest.json` file.
+ @abstract A ``WKWebExtension`` object encapsulates a web extension’s resources that are defined by a `manifest.json`` file.
```
```suggestion
 @abstract A ``WKWebExtension`` object encapsulates a web extension’s resources that are defined by a ``manifest.json`` file.
```

xeenon replied:
Double backticks are really only for symbols. Code formatting should still be single backtick.

jeffmapple replied:
> Double backticks are really only for symbols. Code formatting should still be single backtick.

OK, but that means `manifest.json` has an extra backtick at the end of it.

jeffmapple replied:
```suggestion
 @abstract A ``WKWebExtension`` object encapsulates a web extension’s resources that are defined by a `manifest.json` file.
```

jeffmapple commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
```diff
@@ -518,9 +518,9 @@ WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA)) WK
- Passing a `nil` expiration date will be treated as a distant future date. Only `WKWebExtensionContextPermissionStatusDeniedExplicitly`, `WKWebExtensionContextPermissionStatusUnknown`,
- and `WKWebExtensionContextPermissionStatusGrantedExplicitly` states are allowed to be set using this method.
+ @discussion This method will update ``grantedPermissionMatchPatterns`` and ``deniedPermissionMatchPatterns``. Use this method for changing a single match pattern's status.
+ Passing a `nil` expiration date will be treated as a distant future date. Only ``WKWebExtensionContextPermissionStatusDeniedExplicitly`, `WKWebExtensionContextPermissionStatusUnknown``,
```
```suggestion
 Passing a `nil` expiration date will be treated as a distant future date. Only ``WKWebExtensionContextPermissionStatusDeniedExplicitly``, ``WKWebExtensionContextPermissionStatusUnknown``,
```

Or should these constants be using a single backtick? Either way, the number of backticks are unbalanced.

xeenon replied:
This was a grep error, good catch!

xeenon committed:
Use double-backticks in Web Extension Headers docs for symbols and methods.
https://webkit.org/b/278174
rdar://problem/133961517

Reviewed by Jeff Miller.

Using double backticks is recommended for symbols and methods, and are highlighted
differently in Xcode, and developers can jump to the definition with command-click.

* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataRecord.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataType.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPermission.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.h:

Canonical link: https://commits.webkit.org/282305@main
